lib/state.js

```diff
@@ -468,10 +486,15 @@ StateResolver.prototype.resolveVariable_ = function(name, value) {
   } else if (value.isObject()) {
     data.varTableIndex = this.getVariableIndex_(value);
     var maxProps = this.config_.capture.maxProperties;
-    if (maxProps && maxProps < Object.keys(value.value()).length) {
+    var maxBuffer = this.config_.capture.maxDataSize;
+    var numKeys = Object.keys(value.value()).length;
+
+    if (!isEvaluated && maxProps && maxProps < numKeys) {
       data.status = MESSAGE_TABLE[OBJECT_LIMIT_MESSAGE_INDEX].status;
     }
-
+    else if (isEvaluated && maxBuffer > 0 && maxBuffer < numKeys) {
```

Comment:
Now `maxDataSize` is not being used in comparisons that compare against the number of attributes in an object.  Instead, an evaluated expression is either entirely collected or not entirely collected depending on whether or not the entire contents of that expression would go beyond the limits of `maxDataSize`.